core/src/main/java/com/bankbazaar/tripmanager/manager/TripActivityManager.java

```diff
@@ -0,0 +1,82 @@
+package com.bankbazaar.tripmanager.manager;
+
+import com.bankbazaar.tripmanager.model.TripActivity;
+import com.bankbazaar.tripmanager.repository.TripActivityRepository;
+import org.springframework.beans.factory.ObjectProvider;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.util.Optional;
+
+public class TripActivityManager {
+    @Autowired
+    private TripActivityRepository tripActivityRepository;
+
+    /**
+     * Insert to TripActivity table
+     * @param data
+     */
+    public TripActivity saveTripActivity(TripActivity data)
```

Comment:
Merge save and update into a single function.
The function flow will be
1. Check if incoming data has an id - if it has id then its an update else a create new
2. For update fetch the existing trip details and populate non null fields from new data to existing model

Reply:
This haas been implemented